apps/prairielearn/src/pages/shared/syncHelpers.js

```diff
@@ -34,78 +35,87 @@ module.exports.pullAndUpdate = async function (locals) {
   }
 
   serverJob.executeInBackground(async (job) => {
-    let startGitHash = null;
-    const coursePathExists = await fs.pathExists(locals.course.path);
-    if (!coursePathExists) {
-      // path does not exist, start with 'git clone'
-      job.info('Clone from remote git repository');
-      await job.exec(
-        'git',
-        ['clone', '-b', locals.course.branch, locals.course.repository, locals.course.path],
-        {
-          // Executed in the root directory, but this shouldn't really matter.
-          cwd: '/',
+    const lockName = `coursedir:${locals.course.path}`;
```

Comment:
We are generating this lock name in several places, I think? Should we have a `getCourseSyncLockName(course)` function to make it clear that we always want to use a consistent lock name?

Reply:
Works for me!

Reply:
I opted to do this in a separate PR: #8154